test/extended/util/cli.go

```diff
@@ -0,0 +1,225 @@
+package util
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	. "github.com/GoogleCloudPlatform/kubernetes/test/e2e"
+	. "github.com/onsi/ginkgo"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/admin/policy"
+	"github.com/openshift/origin/pkg/cmd/cli"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	testutil "github.com/openshift/origin/test/util"
+	"github.com/spf13/cobra"
+)
+
+// CLI provides function to call the OpenShift CLI and Kubernetes and OpenShift
+// REST clients.
+type CLI struct {
+	verb            string
+	adminConfigPath string
+	globalArgs      []string
+	commandArgs     []string
+	finalArgs       []string
+	stdout          io.Writer
+	verbose         bool
+	cmd             *cobra.Command
+	kubeFramework   *Framework
+}
+
+// NewCLI initialize the Kubernetes E2E framework, project role bindings.
+// If runInGinko argument is set to true, this function will install BeforeEach
+// hook to Gingo test case that grant project role bindings to a namespace created
+// by the E2E test. This option must be set inside Describe() or Context()
+func NewCLI(project, configPath string, runInGinko bool) *CLI {
```

Comment:
Passing a bool here is pretty kludgy.  Is there any way to auto-detect?  If not, I'd prefer two public methods `NewGinkoCLI` and `NewCLI` to make the intent more clear.